File: carsharing-api/carsharing-api/Entities/Transport.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace carsharing_api.Entities;

[Table("transport")]
[JsonObject(MemberSerialization = MemberSerialization.OptOut)]
public class Transport
{
    public int Id { get; set; }

    [JsonConverter(typeof(StringEnumConverter))]
    public CarType TransportType { get; set; }

    public string CarName { get; set; } = null!;

    public string CarNumber { get; set; } = null!;

    public string CarImagePath { get; set; } = null!;

    public bool IsReserved { get; set; }

    public short GasLevel { get; set; }

    public bool HasInsurance { get; set; }

    public bool IsDoorOpened { get; set; }

    public float Longitude { get; set; }

    public float Latitude { get; set; }
}

File: carsharing-app/app/src/main/java/com/syndicate/carsharing/database/models/Transport.kt
package com.syndicate.carsharing.database.models

import com.fasterxml.jackson.annotation.JsonProperty

data class Transport (
    @JsonProperty("Id") val id: Int,
    @JsonProperty("TransportType") val transportType: String,
    @JsonProperty("CarName") val carName: String,
    @JsonProperty("CarNumber") val carNumber: String,
    @JsonProperty("CarImagePath") val carImagePath: String,
    @JsonProperty("IsReserved") val isReserved: Boolean,
    @JsonProperty("GasLevel") val gasLevel: Int,
    @JsonProperty("InsuranceType") val insuranceType: String,
    @JsonProperty("IsDoorOpened") val isDoorOpened: Boolean,
    @JsonProperty("Longitude") val longitude: Double,
    @JsonProperty("Latitude") val latitude: Double,
    @JsonProperty("GasConsumption") val gasConsumption: Int,
    @JsonProperty("TankCapacity") val tankCapacity: Int
)